task_processing/plugins/smartstack.py

```diff
@@ -0,0 +1,34 @@
+import yaml
+
+
+YOCALHOST_IP = '169.254.255.254'
+TASKPROC_SERVICE_FILE = '/nail/etc/services/task_processing/smartstack.yaml'
+
+
+class MesosLeaderNotFound(Exception):
+    pass
+
+
+class ProxyPortNotFound(Exception):
+    pass
+
+
+def get_mesos_master(cluster, region):
+    """ Finds the yocalhost ip:port for the Mesos master via SmartStack
+    Args customize granularity, defaults to a dev cluster for safety
+    """
+    with open(TASKPROC_SERVICE_FILE, 'r') as istream:
+        taskproc_conf = yaml.load(istream)
+
+    leader_name = 'mesos_leader_paasta_{cluster}_{region}'.format(
```

Comment:
This makes me think this belongs in paasta even more. External code bases can register plugins too, it doesn't have to all be in task proc :-)